client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/InternalFlag.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.client.hotrod.impl.protocol;
+
+/**
+ * Flags which are internal to the protocol and shouldn't be exposed to users.
```

Comment:
This is way too over complicated in my opinion. 

A public flag, which the user controls to decide the granularity of the lifespan/maxidle parameters, with two possible values: SECONDS or MILLISECONDS is more than enough, and you don't have to be doing wacky calculations of whether the value is multiple of certain number or not.

If you go down this route, you're going to have to implement this subtle calculations in all clients you write up. You're complicating the client implementation for no real reason.

I also don't like how this is hidden from the user with an internal flag. If we need to implement it for JCache, we need to expose it externally in a clear fashion. Also remember to think about client implementors, you want to keep things simple and clear for them.